Dockerfile

```diff
@@ -62,6 +63,39 @@ RUN pip3 install --no-cache-dir -r ./docs/requirements.txt
 COPY . /code/listenbrainz
 
 
+#####################################################################################################
+# NOTE: The javascript files are continously watched and compiled using this image in developement. #
+#####################################################################################################
+FROM node:$NODE_VERSION as listenbrainz-frontend-dev
+
+ARG NODE_VERSION
+
+LABEL org.label-schema.vcs-url="https://github.com/metabrainz/listenbrainz-server.git" \
+      org.label-schema.schema-version="1.0.0-rc1" \
+      org.label-schema.vendor="MetaBrainz Foundation" \
+      org.label-schema.name="ListenBrainz Static Builder" \
+      org.metabrainz.based-on-image="node:$NODE_VERSION"
+
+RUN mkdir /code
+WORKDIR /code
+
+COPY package.json package-lock.json /code/
+RUN npm install
+
+COPY webpack.config.js babel.config.js enzyme.config.ts jest.config.js tsconfig.json .eslintrc.js .stylelintrc.js /code/
+
+RUN mkdir /static
+
+#########################################################################
+# NOTE: The javascript files for production are compiled in this image. #
+#########################################################################
+FROM listenbrainz-frontend-dev as listenbrainz-frontend-prod
+
+# Compile front-end (static) files
+COPY ./listenbrainz/webserver/static /static
```

Comment:
Out of curiosity, is there a specific reason for copying and building into the /static folder, rather than what we do in dev  (/code/…) ?
From my understanding since we are copying an arbitrary folder from the image the path shouldn't matter.

The advantage would be that we would only have one path to support instead of different one for dev and prod.

But maybe I'm missing a detail?

Reply:
All the useful files should end up in  /code/listenbrainz/webserver/static/dist once compiled, and we shouldn't need to copy anything else, right?
node modules are all the way at the root /code folder.

I'm thinking `COPY --from=listenbrainz-frontend-prod  /code/listenbrainz/webserver/static/dist /static/dist` might do the trick, but I haven't tried (and my brain stopped working so I shouldn't try that right now :p )

Reply:
That only has the compiled assets, we still want other stuff like /static/img, /static/sound, /static/fonts, robots.txt. But we could explicitly list those I guess.

Reply:
That would make sense to me, and remove some arbitrary confusing path fuckery :) 

Or perhaps we can explicilty ignore the couple of folders we don't need (js, css), if there's an easy syntax for that.

Reply:
I gave it a try in c30da47 , let me know what you think (and if I missed anything !)

Reply:
This copies the contents of `/code/listenbrainz/webserver/static/fonts` to `/static` instead of `/static/fonts` which is likely not desired. I'll split this into multiple COPY --from statements to fix.

Reply:
I put the latest version up on beta for testing as well.

Reply:
Thanks for catching that !
On beta currently all the dropdowns are broken. Fro example in the navbar and on the charts page.
I forgot we need the js/lib folder…